### ISSUE TITLE
Relax SC2182 for bash, when printf -v is used

### ISSUE THREAD

**Issue by jimbobmcgee** [koalaman/shellcheck#2650]
Specifically for *bash*, `printf` has an extra option, `-v` which you can supply a variable name.  This named variable is populated with the result of the `printf` call.  This means you can write `printf -vvar '%s x %s' "$1" "$2"; echo "$var"`.  

Usually, the first arg to `printf` is a format string, so in POSIX shell, an equivalent might be `var="$(printf '%s x %s' "$1" "$1")"; echo "$var"`.

SC2182 fires for *bash* if the first arg is `-vvar`, because `-vvar` does not contain any format string tokens.  However, it is a valid argument to *bash*'s `printf`.

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2182
- My shellcheck version (`shellcheck --version` or "online"): 0.9 (according to https://github.com/vscode-shellcheck/vscode-shellcheck/releases/tag/v0.29.2)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
printf -vvar '%s x %s' "$1" "$2"
echo "$var"
```

#### Here's what shellcheck currently says:
[SC2182](https://www.shellcheck.net/wiki/SC2182) (error): This printf format string has no variables. Other arguments are ignored.

#### Here's what I wanted or expected to see:
no error

**Comment by dotysan** [comment]
It seems to grumble at `printf -vVAR` but not `printf -v VAR` why?

**Comment by jimbobmcgee** [comment]
Not sure if that's a question for me, or even if that is one I can answer.  

From what I can gather (at least in the syntax accepted by *bash*) `printf -v VAR ...` and `printf -vVAR ...` are equivalent (i.e. the space is optional).

I have no Haskell experience, so I can't even begin to work out where that would be checked in the code.
